### ISSUE TITLE
Embedding Layer's mask operation with LSTM Layer Gives Wrong Results When Using a GPU

### ISSUE THREAD

**Issue by amitp-ai** [tensorflow/tensorflow#33834]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04.3
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2.0 GPU version
- Python version: 3.x
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
When using a GPU, the mask operation of the embedding layer (with LSTM layer) gives wrong result. With mask_zero = True, anytime there is an input of 0 in one of the time steps, the output should be same as the previous time step's output. Which is what it does with a CPU. However, using a GPU, it just gives a zero output (which is not the expected behavior).

**Describe the expected behavior**
With mask_zero = True, anytime there is an input of 0 in one of the time steps, the output should be same as the previous time step's output. 

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

########################################
import tensorflow as tf
tf.random.set_seed(0)

inputs = tf.keras.layers.Input(shape=(None,), dtype='int32')
x = tf.keras.layers.Embedding(input_dim=10, output_dim=3, mask_zero=True)(inputs)
outputs = tf.keras.layers.LSTM(2, return_sequences=True)(x)

model = tf.keras.Model(inputs, outputs)
padded_inputs = tf.constant([[1,3,0,0]])
print(model(padded_inputs))

It prints the following:
tf.Tensor(
[[[0.00727878 0.00290126]
  [0.00217528 0.00565582]
  [0.         0.        ]
  [0.         0.        ]]] 

instead of 

tf.Tensor(
[[[0.00727878 0.00290126]
  [0.00217528 0.00565582]
  [0.00217528 0.00565582]
  [0.00217528 0.00565582]]] (which is what it is supposed to print). 
Note that when only using a CPU, it gives the correct result.
########################################

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by gadagashwini** [comment]
Issue is replicating with Tf 2.0.0-gpu on colab. Please take a look at the [gist](https://colab.sandbox.google.com/gist/gadagashwini/c8f7774683b56740684e55d9100ffcda/untitled227.ipynb). Thanks!

**Comment by gtg162y** [comment]
Just wanted to check if and when will this be resolved in the future? Thanks alot for listening!

**Comment by qlzh727** [comment]
Sorry for the late reply, this might take some time on my end. 

Also I think even the numerical result is different here, those different value actually belong to masked timesteps which will be auto ignored by keras. Are you somehow using those values?

**Comment by gadagashwini** [comment]
@gtg162y,
This issue is fixed in Tf==2.1.
Please  find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/8bc5e750d90639563d669741602fccb0/untitled459.ipynb) . 
Please close the issue if it was already resolved for you. Thanks!

**Comment by qlzh727** [comment]
I think this issue is not fixed yet, the gist was run on a CPU env, which doesn't reproduce the issue.

**Comment by gadagashwini** [comment]
Was able to reproduce the issue with Tf 2.1 GPU.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/8abe03c1fea7aac3540225b99b121f68/untitled462.ipynb). Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by Saduf2019** [comment]
I am able to reproduce the issue with [Tf nightly] (2.4.0-dev20200809)(https://colab.research.google.com/gist/Saduf2019/c466f8f21b07450a8438d89e352cc248/untitled346.ipynb) GPU.
Please find the gist here. Thanks!

**Comment by Saduf2019** [comment]
@amitp-ai 
This issue has been fixed in tf-nightly and tf 2.5, please refer tot he [gist here](https://colab.research.google.com/gist/Saduf2019/e5e66d55e18434207cb0f3f699e9f5e7/untitled590.ipynb), can you please upgrade your tf and move this issue to closed status.
